### PR TITLE
framework/media: call src_simple() API for rechanneling in audio manager

### PR DESCRIPTION
**Pull request by zhouxinhe**
- Refine resampleing logic in src_simple().
- Update audio manage code about rechanneling.
- Update mediaplayer example code for easy test.

@Taejun-Kwon @xixidodo

### CODE REVIEW

File: framework/src/media/audio/audio_manager.c [hunk 1]
```diff
@@ -125,17 +125,16 @@ struct audio_device_config_s {
...
+	bool necessary;             // resample / rechannel / format converting is needed
+	uint8_t samprate_types;     // sample rate types supported by card
+	void *buffer;               // pointer to the buffer used for resampling
+	uint32_t buffer_size;       // size in bytes of the  buffer
+	uint32_t frames;            // number of frames in the buffer
+	float ratio;                // sample rate converting ratio
+	src_handle_t handle;        // handle of resampler
+	uint32_t user_sample_rate;  // sample rate from a user
+	uint32_t user_channel;      // channel info from a user
+	uint8_t user_format;        // bytes per sample of user format // TODO format value from the user, format converting will be added later.
```
Comment (SRCN-LiQin) [inline framework/src/media/audio/audio_manager.c]: how about put the uint8 togther, to reduce the size of struct.

File: apps/examples/mediaplayer/mediaplayer_main.cpp [hunk 2]
```diff
@@ -401,10 +404,37 @@ class MediaPlayerController
...
+			cout << " " << offset + i << ". " << filelist[i] << endl;
+		}
+		cout << "====================" << endl;
+
+		int select = userInput(0, offset + filenum - 1);
+		if (select >= offset) {
+			strncpy(TEST_FILE_PATH, filelist[select - offset], sizeof(TEST_FILE_PATH));
+		}
+		for (i = 0; i < filenum; i++) {
+			free(filelist[i]);
```
Comment (SRCN-LiQin) [inline apps/examples/mediaplayer/mediaplayer_main.cpp]: is it necessary to add null check here?

**Comment by seinfra** [comment]
Target : [c5cf549005fc17c79ac3627910aa2a0387d2d6fc] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [c5cf549005fc17c79ac3627910aa2a0387d2d6fc] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [c5cf549005fc17c79ac3627910aa2a0387d2d6fc] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c5cf549005fc17c79ac3627910aa2a0387d2d6fc] - Code Rule Check (C++) OK.

File: apps/examples/mediaplayer/mediaplayer_main.cpp [hunk 3]
```diff
@@ -117,31 +122,14 @@ bool MyMediaPlayer::init(int test)
...
+	case TEST_PCM:
 		makeSource = []() {
-			auto source = std::move(unique_ptr<FileInputDataSource>(new FileInputDataSource("/rom/over_16000.mp3")));
+			auto source = std::move(unique_ptr<FileInputDataSource>(new FileInputDataSource("/rom/44100.pcm")));
+			source->setSampleRate(44100);
+			source->setChannels(2);
 			source->setPcmFormat(AUDIO_FORMAT_TYPE_S16_LE);
 			return std::move(source);
 		};
-		break;
```
Comment (Taejun-Kwon) [inline apps/examples/mediaplayer/mediaplayer_main.cpp]: we added it to test below types before.

Comment (zhouxinhe) [reply apps/examples/mediaplayer/mediaplayer_main.cpp]: all files under /rom/* will be listed out when selecting source.
and then user can select any of them to test, may be a mp3/aac/mp4/opus/... postfix file.
e.g.
```txt
====================
 0. Exit APP
 1. Test PCM
 2. Test BUFFER
 3. Test HTTP
 4. /rom/play.mp4
 5. /rom/mono_44_1k.mp3
 6. /rom/mono_24k.mp3
 7. /rom/res_16k.mp3
====================
```

File: framework/src/media/audio/audio_manager.c [hunk 4]
```diff
@@ -408,95 +407,104 @@ static uint32_t get_closest_samprate(unsigned int origin_samprate, audio_io_dire
...
 
+/*
+ * card: Card informations
+ *       card->resample.buffer tells the frames to resample
+ *       card->resample.frames tells the number of frames to resample
+ * data: Pointer to output buffer to save resampled frames
+ * frames: Capability of the output buffer:`data`
+ * return: On success, returns number of frames resampled(outputted) in the output buffer;
+ *         Otherwise, returns negative value for errors.
+ */
```
Comment (Taejun-Kwon) [inline framework/src/media/audio/audio_manager.c]: frames is not necessary anymore.

**Pull request merged**